R/bootstrap-dea.R

```diff
@@ -0,0 +1,167 @@
+#' @importFrom stats complete.cases median quantile rnorm sd var
+NULL
+
+#' Bootstrap a DEA model
+#'
+#' Run bootstrap on a DEA model to estimate bias corrected efficiency scores and confidence
+#' intervals
+#'
+#' @param dea An object of type pioneer_dea from `compute_dea()`
+#' @param alpha One minus the confidence level required (defaults to 0.05)
+#' @param bw_rule A string with the type of bandwidth rule to be used, or a number with the
+#'   bandwidth parameter. See details.
+#' @param iterations The number of bootstrap iterations to be performed
+#'
+#' @details
+#' The bandwidth argument can be set to either `ucv` for unbiased cross validation,
+#' `silverman` for the Silverman rule, or `scott` for the Scott rule. If you provide a
+#' number, this will be used directly as the bandwidth parameter `h`. This can be useful
+#' to replicate results where `h` is given, such as Simar & Wilson (1998). For most practical
+#' applications of the bootstrap, the default of unbias cross validation is sensible.
+#'
+#' @return A list object of class `pioneer_bootstrap`
+#' @seealso [compute_dea()]
+#'
+#' @export
+bootstrap_dea <- function(dea, alpha = 0.05, bw_rule = 'ucv', iterations = 2000) {
+
+  if (!inherits(dea, 'pioneer_dea')) {
+    msg <- 'You must provide an object of class pioneer_dea. Run compute_dea() first'
+    cli::cli_abort(
+      message = c(
+        msg,
+        'x' = 'You must provide an object of class pioneer_data'
+      ))
+  }
+
+  rts <- attr(dea$model, 'rts')
+  if (!rts %in% c('crs', 'vrs')) {
+    msg <- 'Only constant returns to scale and variable returns to scale are supported'
+    cli::cli_abort(
+      message = c(
+        msg,
+        'x' = 'Only vrs and crs are supported in the rts argument',
+        'i' = 'You supplied {rts}'
+      ))
+  }
```

Comment:
Maybe mention this in the documentation? E.g. under @details.

Reply:
Done. I've also added an example (the same as at the top of this PR).